_posts/2021/2021-02-01-convert-iso-country-code-to-emoji-flag.md

```diff
@@ -0,0 +1,95 @@
+---
+title: "Convert a two character ISO country code to an emoji flag"
+description: "From ISO 3166-1 alpha-2 to lovely emoji flag"
+layout: article
+category: ruby
+image:
+  base: '2021/convert-iso-country-code-to-emoji-flag'
+  alt: "Emoji flags"
+---
+
+We often store countries in our applications as a two-letter code in line with the [ISO 3166-1 alpha-2](https://en.wikipedia.org/wiki/ISO_3166-1_alpha-2) standard, for example "GB" for the United Kingdom or "US" for the United States.
+
+But the people demand emoji and who are we to deny them?
+
+```ruby
+def emoji_flag(country_code)
+  cc = country_code.to_s.upcase
+  return unless cc =~ /\A[A-Z]{2}\z/
+
+  cc.codepoints.map { |c| (c + 127397).chr(Encoding::UTF_8) }.join
+end
+
+> emoji_flag("eu")
+#=> "🇪🇺"
+```
+
+
+## How does this work?
+
+The first two lines are defensive code that makes the method more resilient against different kinds of input.
+
+```ruby
+cc = country_code.to_s.upcase
+```
+
+...ensures you're working with an upper case string. And...
+
+```ruby
+return unless cc =~ /\A[A-Z]{2}\z/
+```
+
+...provides an early return of `nil` when the passed string isn't two captial letters.
+
+You could make an argument to return `""` rather than `nil` to ensure the method always returns a string but although it can be a pain it's quite "ruby-ish" to return `nil`.
```

Comment:
```suggestion
You could make an argument and say we should return `""` rather than `nil` to ensure the method always returns a string, but although it can be a pain, it's more "ruby-ish" to return `nil`.
```